providers/hetznercloud/provider.go

```diff
@@ -145,30 +127,65 @@ func (d *Provider) DeployAgent(ctx context.Context, agent *woodpecker.Agent) err
 		firewalls = append(firewalls, &hcloud.ServerCreateFirewall{Firewall: hcloud.Firewall{ID: fw.ID}})
 	}
 
-	_, _, err = d.client.Server().Create(ctx, hcloud.ServerCreateOpts{
-		Name:     agent.Name,
-		UserData: userdataString,
-		Image:    image,
-		Location: &hcloud.Location{
-			Name: d.location,
-		},
-		ServerType: serverType,
-		SSHKeys:    sshKeys,
-		Networks:   networks,
-		Firewalls:  firewalls,
-		Labels:     d.labels,
+	serverCreateOpts := hcloud.ServerCreateOpts{
+		Name:      agent.Name,
+		UserData:  userdataString,
+		SSHKeys:   sshKeys,
+		Networks:  networks,
+		Firewalls: firewalls,
+		Labels:    d.labels,
 		PublicNet: &hcloud.ServerCreatePublicNet{
 			EnableIPv4: d.enableIPv4,
 			EnableIPv6: d.enableIPv6,
 		},
-	})
-	if err != nil {
-		return fmt.Errorf("%s: Server.Create: %w", d.name, err)
+	}
+
+	for _, raw := range d.serverType {
+		rawType, location, _ := strings.Cut(raw, ":")
+
+		serverType, err := d.LookupServerType(ctx, rawType)
+		if err != nil {
+			return err
+		}
+
+		image, _, err := d.client.Image().GetByNameAndArchitecture(ctx, d.image, serverType.Architecture)
+		if err != nil {
+			return fmt.Errorf("%s: Image.GetByNameAndArchitecture: %w", d.name, err)
+		}
+		if image == nil {
+			return fmt.Errorf("%s: %w: %s", d.name, ErrImageNotFound, d.image)
+		}
+
+		serverCreateOpts.Location = &hcloud.Location{Name: location}
```

Comment:
So location name is allowed to be empty here?

Reply:
See my comment. Yes it is allowed to be empty now.